pandas/tests/frame/methods/test_replace.py

```diff
@@ -1681,3 +1681,31 @@ def test_replace_bytes(self, frame_or_series):
         expected = obj.copy()
         obj = obj.replace({None: np.nan})
         tm.assert_equal(obj, expected)
+
+    @pytest.mark.parametrize(
+        "value, to_replace, expected, dtype",
+        [
+            (
+                [1.0, 2.0, 3.999, 4.4],
+                {1.0: 9},
```

Comment:
Since different paths are exercised by `.replace(1.0, 9.0)` and `.replace({1.0, 9.0})`, can the test be parameterized such that those cases can be tested at the same time?